File: Chapter09-Fragments/93/app/src/main/java/com/example/wil/a93/MainActivity.java
package com.example.wil.a93;

import android.graphics.Color;
import android.net.Uri;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity
        implements BlankFragment.OnFragmentInteractionListener
{
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Bundle arguments = new Bundle();

        String param1, param2 ;
        param1 = "param1";
        param2 = "param2";
        // arguments.putString(BlankFragment.ARG_PARAM1, param1)
        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();
        BlankFragment fragment1 = BlankFragment.newInstance(param1,param2) ;
        ft.replace(R.id.topic_detail_container_1,fragment1);
        BlankFragment fragment2 = BlankFragment.newInstance(param1,param2) ;
        ft.replace(R.id.topic_detail_container_2,fragment2);
        ft.commit();


    }
    @Override
    public void onFragmentInteraction(Uri uri){
        //you can leave it empty
    }
}

File: keypad/app/src/main/java/com/example/wil/keypad/MainActivity.kt
package com.example.wil.keypad

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import bsh.Interpreter

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
       btnEq.setOnClickListener { evalInput() }
    }
    fun evalInput(){
        var i = Interpreter()
        var calculation: String? = editText.text.toString()
        var result:Int?
        i.set ("calculation", calculation)
       result= i.eval(calculation) as Int

        textView2.text=result.toString()

    }
}
